src/main/java/org/sunbird/karmapoints/service/KarmaPointsServiceImpl.java

```diff
@@ -71,11 +71,16 @@ public Map<String, Object> fetchKarmaPointsUserCourse(String userId, Map<String,
         Map<String, Object> propertyMap = new HashMap<>();
         String key = userId +"|"+cntxType+"|"+ cntxtId;
         propertyMap.put(Constants.DB_COLUMN_USER_KARMA_POINTS_KEY, key);
-        Map<String, Object> userCourseKpList = cassandraOperation.getRecordsByProperties(Constants.KEYSPACE_SUNBIRD,
-                Constants.TABLE_KARMA_POINTS_LOOK_UP, propertyMap, null, Constants.DB_COLUMN_USER_KARMA_POINTS_KEY);
+        List<Map<String, Object>> userCourseKpList = cassandraOperation.getRecordsByProperties(Constants.KEYSPACE_SUNBIRD,
```

Comment:
Let's use the following method to access this table.. the existing one will add "allow filtering" - which will reduce the performance. We do provide the primary key to read the row, so "allow filtering" is not required.

List<Map<String, Object>> getRecordsByPropertiesWithoutFiltering(String keyspaceName, String tableName,
			Map<String, Object> propertyMap, List<String> fields);